addons/website/models/assets.py

```diff
@@ -44,6 +60,60 @@ def make_scss_customization(self, url, values):
                 'copyright-gradient': 'null',
             })
 
+        google_local_fonts = values.get('google-local-fonts')
+        if google_local_fonts and google_local_fonts != 'null':
+            # "('font x': 45, 'font y': '')" -> "{'font x': 45, 'font y': ''}"
+            google_local_fonts = safe_eval.safe_eval('{%s}' % google_local_fonts[1:-1])
+            updated_font = False
+            headers_woff2 = {
+                'user-agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.41 Safari/537.36',
+            }
+            for font_name in google_local_fonts:
+                if not google_local_fonts[font_name]:
+                    font_family_attachments = IrAttachment
+                    font_content = requests.get(
+                        f'https://fonts.googleapis.com/css?family={font_name}&display=swap',
+                        headers=headers_woff2,
+                    ).content.decode()
+
+                    def fetch_google_font(src):
+                        statement = src.group()
+                        url, font_format = re.findall(r'src: url\(([^\)]+)\) (.+)', statement)[0]
+                        req = requests.get(url, headers=headers_woff2)
+                        # https://fonts.gstatic.com/s/modak/v18/EJRYQgs1XtIEskMB-hRp7w.woff2
+                        # -> s-modak-v18-EJRYQgs1XtIEskMB-hRp7w.woff2
+                        name = url_parse(url).path.lstrip('/').replace('/', '-')
+                        attachment = IrAttachment.create({
+                            'name': f'google-font-{name}',
+                            'type': 'binary',
+                            'datas': base64.b64encode(req.content),
+                            'public': True,
+                        })
+                        nonlocal font_family_attachments
+                        font_family_attachments += attachment
+                        return 'src: url(/web/content/%s/%s) %s' % (
+                            attachment.id,
+                            name,
+                            font_format,
+                        )
+
+                    font_content = re.sub(r'src: url\(.+\)', fetch_google_font, font_content)
+
+                    attach_font = IrAttachment.create({
+                        'name': f'{font_name} (google-font)',
+                        'type': 'binary',
+                        'datas': base64.encodebytes(font_content.encode()),
+                        'mimetype': 'text/css',
+                        'public': True,
+                    })
+                    google_local_fonts[font_name] = attach_font.id
+                    font_family_attachments.original_id = attach_font.id
+                    updated_font = True
+
+            if updated_font:
```

Comment:
I would remove that "updated_font" check: you already only write on this... if the caller asked for it, right? :thinking: `google_local_fonts = values.get('google-local-fonts')`